### ISSUE TITLE
Why does my device config look different?

### ISSUE THREAD

**Issue by StrandedAlien** [dgatf/logic_analyzer_rp2040#1]
Hola Amigo, thank you very much for sharing your code. Awesome work. A 200 MHz LA with triggers for EVERYONE. Gracias! 

Do you know why my device config screen looks different than yours? It lacks of the clock edge selector and the external clock mode ticker. This is on Manjaro Linux up-to-date and i installed your libsigrok over the pre-installed Manjaro standard. I was a bit puzzled cause you recommended to clone the libsigrok original repository. Did not make sense to me, so i downloaded your libsigrok as a zip and compiled and installed from here. What did i possibly do wrong?
![My_config](https://github.com/dgatf/logic_analyzer_rp2040/assets/89656885/a0cdf8f5-057a-4c2a-aee3-50c366b7903f)
![pulseview_device_config](https://github.com/dgatf/logic_analyzer_rp2040/assets/89656885/d0a0dbd6-34e1-4ec7-8151-3bc56884afcf)

**Comment by dgatf** [comment]
That's weird. The fork changes only one character to change to little endian. I've tested with and without the mod and I see the correct config window.

I'm running Ubuntu 22.04.3-6.2.0-34-generic.

Attached the output for libsigrok `./configure` and `pulseview -V`:

[libsigrok_configure.txt](https://github.com/dgatf/logic_analyzer_rp2040/files/12916845/libsigrok_configure.txt)
[pulseview_version.txt](https://github.com/dgatf/logic_analyzer_rp2040/files/12916778/pulseview_version.txt)

Anyway you are missing _Clock edge_ and _External clock mode_, which are not implemented in Logic Analyzer. Only RLE and Pre-trigger capture ratio are implemented and those appears.

Does the _official_ libsigrok from your distribution show the correct config window?

**Comment by StrandedAlien** [comment]
Thanks for the fast answer. I tried using the latest Appimage from https://sigrok.org/download/binary/pulseview/PulseView-NIGHTLY-x86_64.AppImage
of 	2023-10-16 08:00
It gives me this:
![using latest nighlty Appimage](https://github.com/dgatf/logic_analyzer_rp2040/assets/89656885/10c1f1d7-7668-47ee-98f4-2aceea2f8636)

But how to i integrate your update now? Is it crucial to do so, or is it just better and not really necessary? What difference does it make?

**Comment by dgatf** [comment]
You have already patched _libsigrok_. There is no need to patch _Pulseview_

The issue is that if you install the fork for _libsigrok_, which is based in latest code, you may also need to install an updated version of _Pulseview_ 

If you use to official libsigrok, the only noticeable difference is an additional sample size with a enormous big size (around 1000M samples). Logic analyzer sends a maximum of 100K samples 

![pulseview_issue](https://github.com/dgatf/logic_analyzer_rp2040/assets/33811722/3178d33a-f12e-4866-9319-307c9120c6b0)

If you dont want to use the appimage, you can either install libsigrok from your distribution repositories or build pulseview and install it

**Comment by StrandedAlien** [comment]
I do not get this to work. It always triggers when running, it does not wait for a changing edge. That happens even when nothing at all is connected. It starts sampling right away after pushing run. I used a Saleae 16 clone for comparison, which works as expected.

**Comment by dgatf** [comment]
Check that you have downloaded the latest binary version: [logic_analyzer.uf2](https://drive.google.com/file/d/1JJ5_vQVWJ0RjmcC2NYjNZ1x6iYy9dHWl/view?usp=drive_link)

I've uploaded a video that shows how to proceed and that triggers are working fine. Sorry for the bad quality of the video.

[Logic analyzer video](https://youtu.be/_8J2ST2amZ0)

**Comment by StrandedAlien** [comment]
Thanks a lot for caring. It works now. AWESOME!!!! You are the boss! And even over sump using pio.  Incredible.
I was so stupid and connected TTL-level output without voltage dividers. RP2040 does not like it at all.
This is so cool. Terimah kasih!

**Comment by StrandedAlien** [comment]
I have tested it using a second pico with an oscilloscope and square wave generator software.
https://github.com/jklomp/pico-RP2040-oscilloscope

Just drop the uf2 on the second pico. Connect GND and signal and start the html for the interface.
I had to use Chromium with developer options enabled, as other browsers (Brave, Ff) could not open the serial port.
Also use a recent Pulseview Appimage as the one from Arch repo did not show the external trigger option ( though not needed here)

![Freescope](https://github.com/dgatf/logic_analyzer_rp2040/assets/89656885/fe71f680-623d-49e8-95a7-2f81b0eaa34e)

![LA-Scope_Timing](https://github.com/dgatf/logic_analyzer_rp2040/assets/89656885/d2605c7c-d987-4890-b6b6-cc108b0737eb)
